packages/start/src/config/index.ts

```diff
@@ -204,31 +218,42 @@ export function defineConfig(
                 ...injectDefineEnv('TSS_SERVER_BASE', serverBase),
                 ...injectDefineEnv('TSS_API_BASE', apiBase),
               },
+              ssr: mergeSsrOptions([
+                viteConfig.userConfig.ssr,
+                ssrViteConfig.userConfig.ssr,
+                {
+                  noExternal: ['@tanstack/start', 'tsr:routes-manifest'],
+                  external: ['@vinxi/react-server-dom/client'],
+                },
+              ]),
+              optimizeDeps: {
+                entries: [],
+                ...(viteConfig.userConfig.optimizeDeps || {}),
+                ...(ssrViteConfig.userConfig.optimizeDeps || {}),
+                // include: ['@tanstack/start/server-runtime'],
+              },
+            }),
+            TanStackStartVitePlugin({
+              env: 'server',
+            }),
+            TanStackServerFnsPlugin.ssr,
+            TanStackRouterVite({
```

Comment:
When using the output from `@tanstack/router-plugin`, there's a toggle named [`enableRouteGeneration`](https://github.com/TanStack/router/blob/40f9f49e4b1648884da3aa1addbe267194d938ee/packages/router-plugin/src/core/router-generator-plugin.ts#L70) (that defaults to `true`) which controls whether the router-generator is run thereby running the file-system reads and writes. It has no bearing on the automatic code splitting.

Would it make sense to only have this flag active on a single vinxi router? and therefore disabled on the others to save having duplicated reads and writes on the user's machine.